File: feature/base/src/main/java/net/twisterrob/android/view/color/swatches/EmptySwatch.java
package net.twisterrob.android.view.color.swatches;

import android.graphics.Canvas;

public class EmptySwatch extends Swatch {
	private int color;

	@Override
	public int getCurrentColor() {
		return color;
	}

	@Override
	public void setCurrentColor(int color) {
		this.color = color;
	}

	@Override
	public int findColor(int trackedArea, float x, float y) {
		return color;
	}

	@Override
	public void setCurrentArea(int areaCode) {
	}

	@Override
	public void draw(Canvas canvas) {
		canvas.drawColor(color);
	}
}

File: feature/base/src/main/kotlin/net/twisterrob/android/view/color/swatches/EmptySwatch.kt
package net.twisterrob.android.view.color.swatches

import android.graphics.Canvas
import android.graphics.Color
import androidx.annotation.ColorInt

class EmptySwatch : Swatch() {

	@ColorInt
	override var currentColor: Int = Color.TRANSPARENT

	@ColorInt
	override fun findColor(area: AreaCode, x: Float, y: Float): Int {
		return currentColor
	}

	override fun draw(canvas: Canvas) {
		canvas.drawColor(currentColor)
	}
}
